models/migrations/v1_19/v240.go

```diff
@@ -0,0 +1,14 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_19 //nolint
+
+import (
+	"xorm.io/xorm"
+)
+
+func AddArchivedUnixForRepository(x *xorm.Engine) error {
+	// all previous tokens have `all` and `sudo` scopes
+	_, err := x.Exec("UPDATE repository SET archived_unix = 0 WHERE archived_unix IS NULL")
```

Comment:
Hmm, this doesn't seem like a good migration/architecture:
1. The new attribute must first be added to the database.
2. I don't think we should set `= 0` manually. Either use `DEFAULT 0` in the definition, or don't set it at all when not needed. I think `NULL` is in this case an acceptable value as repos cannot be archived at all.
3. I don't think we should use `0 = Unarchived` if we store `IsArchived` too. 
4. This line is wrong as it shouldn't only use `IS NULL` as condition, but `AND is_archived = true` 
So, something like
```go
err := x.Where("archived_unix IS NULL AND is_archived = ?", true).SetExpr("archived_unix", timeutil.TimeStampNow()).Update(new(Repo))
```
This way, already archived repos will simply be "archived when the gitea version is upgraded"

Reply:
I now use `DEFUALT 0`, so this is no longer needed

Reply:
Hmm, but this means already archived repos will be shown as "has no archived date", so the go code in my suggestion above is still needed.
Or should already archived repos have that inconsistency?

Reply:
Tey should have this inconsistency. The Archive Date should say, when I repo is archived. This means it shows when the work on the project has stopped. Setting the upgrade date  as archive date for Repos, which are archived for Years is confusing.

Reply:
I know, but I think it is better than not displaying an archived date at all.
It has a benefit:
If we do that, we can be sure that `archived_unix == 0 => is_archived = false`, which we otherwise can't.

Reply:
For what do we need  `archived_unix == 0 => is_archived = false` ?

Reply:
At the moment: It isn't needed for anything as this PR only introduces it.
However, as I've been continuously taught:
Old assumptions are easily forgotten in large (open-source) projects over the years.
I'm fairly certain that someone will at some point rely on exactly this assumption without knowing that it doesn't apply.

Reply:
Why not adding a comment e.g. is zero if repo is not archived or archive date is unknown

Reply:
Yup, that works too.